### ISSUE TITLE
E3682 for StorageEncrypted on AWS::RDS::DBInstance

### ISSUE THREAD

**Issue by pbrisbin** [aws-cloudformation/cfn-lint#4068]
### CloudFormation Lint Version

1.32.3

### What operating system are you using?

Arch Linux, Ubuntu

### Describe the bug

```yaml
  AuroraPrimaryInstance:
    Type: AWS::RDS::DBInstance
    Properties:
      EngineVersion: "16.2"
      StorageEncrypted: !Ref "StorageEncrypted"
      DBClusterIdentifier: !Ref "AuroraCluster"
      PubliclyAccessible: false
      AutoMinorVersionUpgrade: false
      DBSubnetGroupName: !ImportValue "networking-RdsSubnetGroup"
      MonitoringRoleArn: !GetAtt "EnhancedMonitoringRole.Arn"
      MonitoringInterval: 15
      Engine: aurora-postgresql
      DBInstanceClass: !Ref "PrimaryInstanceClass"
      DBInstanceIdentifier: !Ref "PrimaryInstanceIdentifier"
      EnablePerformanceInsights: true
      DBParameterGroupName: !Ref "AWS::NoValue"
      Tags:
        - Key: role
          Value: rds-primary
```

```
E3682 Additional properties are not allowed (['StorageEncrypted'])
templates/progress-aurora-cluster.yaml:134:5
```

Removing the "additional" property indeed triggers change,

```
Changes for stacks/xxxx.freckle/us-east-1/progress/prod/aurora-cluster.yaml:
  Modify AuroraPrimaryInstance (AWS::RDS::DBInstance) progress-db-aurora-primary-prod
    Replacement: True
    Scope: Properties
    Details:
      * DirectModification in Properties (StorageEncrypted), recreation Always
```

### Expected behavior

cfn-lint should not report this property as additional/unnecessary, since CloudFormation appears to want to use it.

### Reproduction template

```yaml
  AuroraPrimaryInstance:
    Type: AWS::RDS::DBInstance
    Properties:
      EngineVersion: "16.2"
      StorageEncrypted: !Ref "StorageEncrypted"
      DBClusterIdentifier: !Ref "AuroraCluster"
      PubliclyAccessible: false
      AutoMinorVersionUpgrade: false
      DBSubnetGroupName: !ImportValue "networking-RdsSubnetGroup"
      MonitoringRoleArn: !GetAtt "EnhancedMonitoringRole.Arn"
      MonitoringInterval: 15
      Engine: aurora-postgresql
      DBInstanceClass: !Ref "PrimaryInstanceClass"
      DBInstanceIdentifier: !Ref "PrimaryInstanceIdentifier"
      EnablePerformanceInsights: true
      DBParameterGroupName: !Ref "AWS::NoValue"
      Tags:
        - Key: role
          Value: rds-primary
```

Apologies, this is not a standalone example.

**Comment by kddejong** [comment]
Yea this is a weird trick and debating on how to handle it.  Indeed this property is ignored when using aurora but since CloudFormation has this flagged as requiring replacement its triggering the replacement.  Need to discuss with a few people on this one.

**Comment by kddejong** [comment]
I'm going to remove this property from the check.

The docs are a little off here.  While the value is ignored it must match the cluster or you get: 
`# If you specify storageEncrypted for instance that will be part of cluster it should match cluster setting; requested: true, cluster: false` So you can specify it, it just has to match the cluster level setting, and if you don't specify it will use the cluster setting

**Comment by pbrisbin** [comment]
Thank you!
